### PR TITLE
add a clean to ensure we do not lock files for a windows build

### PR DESCRIPTION
**Pull request by olamy**
cherry-pick of https://github.com/apache/maven-gh-actions-shared/commit/6fb412f2817e5c91dd530405f4568204e4fbea6e

### CODE REVIEW

File: .github/workflows/maven-verify.yml [hunk 1]
```diff
@@ -184,10 +184,17 @@ jobs:
...
+      - name: Clean Ensuring no file handle remains open on windows
+        run: ./mvnw clean --errors --batch-mode --show-version           
 
       - name: Build Maven Site
         run: ./mvnw --errors --batch-mode --show-version ${{ inputs.maven-args }} ${{ inputs.ff-site-goal }}
         if: inputs.ff-site-run
+        
+      - name: Clean Ensuring no file handle remains open on windows
+        if: steps.should-run.conclusion == 'success'
+        run: ./mvnw clean --errors --batch-mode --show-version
```
Comment (slawekjaranowski) [inline .github/workflows/maven-verify.yml:197]: put it after `Show free disk space` we need files for `Upload Maven Site` and for `Upload artifact on failure`

File: .github/workflows/maven-verify.yml [hunk 2]
```diff
@@ -184,10 +184,17 @@ jobs:
 
       - name: Build with Maven
         run: ./mvnw --errors --batch-mode --show-version ${{ inputs.maven-args }} ${{ inputs.ff-goal }}
+        
+      - name: Clean Ensuring no file handle remains open on windows
+        run: ./mvnw clean --errors --batch-mode --show-version
```
Comment (slawekjaranowski) [inline .github/workflows/maven-verify.yml:189]: We have two `clean` in `fail-fast-build` - one should be in `verify` jobs

File: .github/workflows/maven-verify.yml (hunk 1 above)
Comment (olamy) [reply .github/workflows/maven-verify.yml]: yup we don't know if the basic build or the site build is locking.
but that would be nit picking

File: .github/workflows/maven-verify.yml [hunk 3]
```diff
@@ -207,6 +204,10 @@ jobs:
         if: always()
         run: df -h
         shell: bash
+        
+      - name: Clean Ensuring no file handle remains open on windows
+        if: steps.should-run.conclusion == 'success'
```
Comment (slawekjaranowski) [inline .github/workflows/maven-verify.yml:209]: step `should-run` only exist in `verify` jobs

File: .github/workflows/maven-verify.yml [hunk 4]
```diff
@@ -182,13 +182,10 @@ jobs:
         run:
           mvn --errors --batch-mode --show-version org.apache.maven.plugins:maven-wrapper-plugin:3.1.0:wrapper "-Dmaven=${{ inputs.ff-maven }}"
 
-      - name: Build with Maven
-        run: ./mvnw --errors --batch-mode --show-version ${{ inputs.maven-args }} ${{ inputs.ff-goal }}
-
```
Comment (slawekjaranowski) [inline .github/workflows/maven-verify.yml:187]: should be not removed

Comment (olamy) [reply .github/workflows/maven-verify.yml]: sorry grrhhh %#%@%^#@*^%# gh web editor (pardon my french)

File: .github/workflows/maven-verify.yml [hunk 5]
```diff
@@ -188,7 +188,7 @@ jobs:
       - name: Build Maven Site
         run: ./mvnw --errors --batch-mode --show-version ${{ inputs.maven-args }} ${{ inputs.ff-site-goal }}
         if: inputs.ff-site-run
-
+
```
Comment (slawekjaranowski) [inline .github/workflows/maven-verify.yml:191]: additional whitespace - not important.

**Review by slawekjaranowski — APPROVED** [review]

**Pull request merged**